### ISSUE TITLE
Wooden Staircases?

### ISSUE THREAD

**Issue by zohozer** [s-leger/blenderPolygons#2]
Hi. I just discovered this add-on and I'm very interested in creating some wood staircases.

It is possible to develop this add-on so also be able to generate wooden staircases with Steps, Risers, Strings, Newel Post, Spindles and Handrails? Also will be great to generate automatic boolean operations between the components.

Something like this:

http://www.stairplan.com/Assets/images/ideas/prescott-staircase.jpg

http://images.homestylecentral.com/2016/03/3-shutterstock_67272868-r80.jpg

I am a Staircase designer and I can provide further assistance/feedback if needed.

**Comment by s-leger** [comment]
Hi, 
Thank you for testing this addon, and for your feedback.

Stairs is one of the most recent part of the Archipack, along walls, and not finished yet, but does work well for metal, concrete and modern wood stairs. He definitly need some more love on the Balustrade side (allowing baluster between posts). 

He's able to generate String and simple Handrails, by enabling either left and/or right Ladder, increasing the number option, and use the ladder panel to setup the size and position according your needs.
(Ladder basically are handrails, but decreasing altitude and playing with offset and size may allow to build Strings - cables - anything else with linear altitude inerpolation)

Automatic boolean within a single object is not possible. The only way is to be able to generate multiple stairs, each one with her own part, and then make boolean with resulting objects.

**Comment by zohozer** [comment]
> Automatic boolean within a single object is not possible. The only way is to be able to generate multiple stairs, each one with her own part, and then make boolean with resulting objects.

Yes you are right, but I'm thinking that the script can be modified to generate one different object per each different element of the stair. So for example, if I have a stair with twelve steps, one landing, six strings and four posts, the result to be something like this: step1-(to)-step12, land1, riser1-(to)-riser14, str1-(to)-str6, post1-(to)-post6, each one as a separate object, so after that to be much easier to make the necessary boolean operations between the different parts.

For better understanding I will attach a blend file as an example of an imported staicase model, exported as DAE from a staicase design software. Into this 3D model it is missing the cuts of the steps and risers into strings and posts, but they are present into the mechanisation files. This model it is exported only for visualisation purposes on Sketchfab.
[Timber_DAE_Staircase.zip](https://github.com/s-leger/blenderPolygons/files/949860/Timber_DAE_Staircase.zip)

**Comment by s-leger** [comment]
Separating object by parts may be possible by setting materials for each part you want to make standalone then in edit mode, hit p and pick separate by material (or by loose parts)

Look like your stairs strings altitudes are not linear, but curve following steps. 
I'll take a look at make this kind of shape for strings.

As a side note:  
This set of addons was not designed to make BIM-like modelling nor print ready solid modelling.  There are tools like FreeCAD for such needs.
Not impossible, but simply not the target as features will be limited to what's needed to do quick arch modelling for visualisation.
